lib/mongoid/indexable/specification.rb

```diff
@@ -63,39 +63,73 @@ def name
 
       private
 
-      # Normalize the spec, in case aliased fields are provided.
+      # Normalize the spec in place, in case aliased fields are provided.
       #
       # @api private
       #
       # @example Normalize the spec.
-      #   specification.normalize_key(name: 1)
+      #   specification.normalize_aliases!(name: 1)
       #
-      # @param [ Hash ] key The index key(s).
+      # @param [ Hash ] spec The index specification.
       #
       # @return [ Hash ] The normalized specification.
-      def normalize_key(key)
-        normalized = {}
-        key.each_pair do |name, direction|
-          normalized[klass.database_field_name(name).to_sym] = direction
+      def normalize_aliases!(spec)
+        return unless spec.is_a?(Hash)
+
+        spec.transform_keys! do |name|
+          klass.database_field_name(name).to_sym
         end
-        normalized
       end
 
-      # Normalize the index options, if any are provided.
+      # Normalize the index options in place.
       #
       # @api private
       #
       # @example Normalize the index options.
-      #   specification.normalize_options(unique: true)
+      #   specification.normalize_options!(unique: true)
       #
-      # @param [ Hash ] opts The index options.
+      # @param [ Hash ] options The index options.
       #
       # @return [ Hash ] The normalized options.
-      def normalize_options(opts)
-        options = {}
-        opts.each_pair do |option, value|
-          options[MAPPINGS[option] || option] = value
+      def normalize_options!(options)
+
+        options.transform_keys! do |option|
+          option = option.to_sym
+          MAPPINGS[option] || option
+        end
+
+        %i[weights wildcard_projection].each do |key|
+          normalize_aliases!(options[key])
+        end
+
+        recursive_normalize_conditionals!(options[:partial_filter_expression])
+
+        options
+      end
+
+      # TODO: DOCS.
```

Comment:
Reminder:

Reply:
Yep, waiting on Oleg's reply to this comment: https://github.com/mongodb/mongoid/pull/5267#issuecomment-1207218570